<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1275</width>
    <height>1032</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Hydrus Station</string>
  </property>
  <property name="windowIcon">
   <iconset resource="resources.qrc">
    <normaloff>:/res/logo-v2-standalone.png</normaloff>:/res/logo-v2-standalone.png</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QWidget" name="mapContainer" native="true">
      <property name="styleSheet">
       <string notr="true">#mapContainer{border: 1px solid rgb(196, 197, 198);
}</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <property name="styleSheet">
    <string notr="true">#mainToolBar{background:transparent; border: none}</string>
   </property>
   <property name="toolButtonStyle">
    <enum>Qt::ToolButtonTextUnderIcon</enum>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionServer_Setup"/>
   <addaction name="actionDisconnect"/>
   <addaction name="separator"/>
   <addaction name="actionNavigate"/>
   <addaction name="actionOpen_Log"/>
   <addaction name="actionRemote_Control"/>
   <addaction name="separator"/>
   <addaction name="actionCenter_Map"/>
   <addaction name="actionAddRoute"/>
   <addaction name="actionDelete_Route"/>
   <addaction name="actionSave_Routes"/>
   <addaction name="separator"/>
   <addaction name="actionWarmstart_GPS"/>
   <addaction name="actionSend_Command"/>
   <addaction name="separator"/>
   <addaction name="actionAbout"/>
  </widget>
  <widget class="QDockWidget" name="routeDock">
   <property name="sizePolicy">
    <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="minimumSize">
    <size>
     <width>250</width>
     <height>560</height>
    </size>
   </property>
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="windowTitle">
    <string>Route</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QFormLayout" name="formLayout">
     <item row="0" column="0" colspan="2">
      <widget class="QComboBox" name="routesCombo">
       <property name="font">
        <font>
         <weight>75</weight>
         <bold>true</bold>
        </font>
       </property>
       <property name="currentText">
        <string/>
       </property>
       <property name="currentIndex">
        <number>-1</number>
       </property>
      </widget>
     </item>
     <item row="1" column="0" colspan="2">
      <widget class="QGroupBox" name="groupBox">
       <property name="title">
        <string>Base Location</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QLineEdit" name="routeBaseLocation">
          <property name="readOnly">
           <bool>true</bool>
          </property>
          <property name="placeholderText">
           <string>No base location</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="routeSetStartGPS">
          <property name="text">
           <string>Set to current GPS</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="routeSetStartCursor">
          <property name="text">
           <string>Set to map cursor</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="routeReturnToBase">
          <property name="text">
           <string>Return to base when done</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item row="2" column="0" colspan="2">
      <widget class="QGroupBox" name="groupBox_2">
       <property name="title">
        <string>Waypoints</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="1" column="3">
         <widget class="QToolButton" name="wayDel">
          <property name="toolTip">
           <string>Delete waypoint</string>
          </property>
          <property name="text">
           <string>...</string>
          </property>
          <property name="icon">
           <iconset theme="edit-delete">
            <normaloff>.</normaloff>.</iconset>
          </property>
         </widget>
        </item>
        <item row="0" column="0" colspan="8">
         <widget class="QListView" name="routeWaypointsView"/>
        </item>
        <item row="1" column="2">
         <widget class="QToolButton" name="wayAdd">
          <property name="toolTip">
           <string>Add new waypoint</string>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset theme="list-add">
            <normaloff>.</normaloff>.</iconset>
          </property>
          <property name="popupMode">
           <enum>QToolButton::InstantPopup</enum>
          </property>
         </widget>
        </item>
        <item row="1" column="7">
         <widget class="QToolButton" name="wayClear">
          <property name="toolTip">
           <string>Clear all waypoints</string>
          </property>
          <property name="text">
           <string>...</string>
          </property>
          <property name="icon">
           <iconset theme="edit-clear">
            <normaloff>.</normaloff>.</iconset>
          </property>
         </widget>
        </item>
        <item row="1" column="5">
         <widget class="QToolButton" name="wayUp">
          <property name="toolTip">
           <string>Move waypoint up</string>
          </property>
          <property name="text">
           <string>...</string>
          </property>
          <property name="icon">
           <iconset theme="go-up">
            <normaloff>.</normaloff>.</iconset>
          </property>
         </widget>
        </item>
        <item row="1" column="6">
         <widget class="QToolButton" name="wayDown">
          <property name="toolTip">
           <string>Move waypoint down</string>
          </property>
          <property name="text">
           <string>...</string>
          </property>
          <property name="icon">
           <iconset theme="go-down">
            <normaloff>.</normaloff>.</iconset>
          </property>
         </widget>
        </item>
        <item row="1" column="4">
         <widget class="QToolButton" name="waySet">
          <property name="toolTip">
           <string>Set waypoint position</string>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset theme="zoom-fit-best">
            <normaloff>.</normaloff>.</iconset>
          </property>
          <property name="popupMode">
           <enum>QToolButton::InstantPopup</enum>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item row="3" column="0" colspan="2">
      <widget class="QGroupBox" name="groupBox_3">
       <property name="title">
        <string>Info</string>
       </property>
       <layout class="QFormLayout" name="formLayout_2">
        <property name="labelAlignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
        <item row="3" column="0">
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Estimated time</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Total Length</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QLabel" name="routeInfoLen">
          <property name="text">
           <string>-</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QLabel" name="routeInfoEstTime">
          <property name="text">
           <string>-</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Last Update</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLabel" name="routeInfoUpd">
          <property name="text">
           <string>-</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="statusDock">
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="windowTitle">
    <string>Status</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="StatusWidget" name="statusDockContents"/>
  </widget>
  <widget class="QDockWidget" name="logDock">
   <property name="windowTitle">
    <string>Log</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="logDockWidget">
    <layout class="QGridLayout" name="gridLayout">
     <item row="0" column="1" rowspan="3">
      <widget class="QTextEdit" name="textEdit">
       <property name="horizontalScrollBarPolicy">
        <enum>Qt::ScrollBarAlwaysOff</enum>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="0" column="2">
      <widget class="QPushButton" name="logSaveBtn">
       <property name="text">
        <string>Save</string>
       </property>
      </widget>
     </item>
     <item row="1" column="2">
      <widget class="QPushButton" name="logClearBtn">
       <property name="text">
        <string>Clear</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="rcDock">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="floating">
    <bool>false</bool>
   </property>
   <property name="windowTitle">
    <string>Remote Control</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="RCWidget" name="rcDockContents"/>
  </widget>
  <action name="actionNavigate">
   <property name="icon">
    <iconset theme="media-playback-start">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Navigate</string>
   </property>
   <property name="toolTip">
    <string>Start navigation with the selected route</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionDisconnect">
   <property name="icon">
    <iconset theme="network-disconnect">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Disconnect</string>
   </property>
  </action>
  <action name="actionRemote_Control">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset theme="network-wireless">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Remote Control</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="actionAddRoute">
   <property name="icon">
    <iconset theme="list-add">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Add Route</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionDelete_Route">
   <property name="icon">
    <iconset theme="list-remove">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Delete Route</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Del</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="icon">
    <iconset theme="help-about">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>About</string>
   </property>
   <property name="shortcut">
    <string>F1</string>
   </property>
  </action>
  <action name="actionServer_Setup">
   <property name="icon">
    <iconset theme="configure">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Configure</string>
   </property>
  </action>
  <action name="actionCenter_Map">
   <property name="icon">
    <iconset theme="view-restore">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Center Map</string>
   </property>
  </action>
  <action name="actionSave_Routes">
   <property name="icon">
    <iconset theme="document-save">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Save Routes</string>
   </property>
   <property name="toolTip">
    <string>Save the current route set</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSimulation_Mode">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset theme="clock">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Simulation Mode</string>
   </property>
  </action>
  <action name="actionSend_Command">
   <property name="icon">
    <iconset theme="system-run">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Command</string>
   </property>
  </action>
  <action name="actionWarmstart_GPS">
   <property name="icon">
    <iconset theme="gps">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Warmstart GPS</string>
   </property>
  </action>
  <action name="actionOpen_Log">
   <property name="icon">
    <iconset theme="text-plain"/>
   </property>
   <property name="text">
    <string>Open Log</string>
   </property>
   <property name="toolTip">
    <string>Open a log file locally or from the drone</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>StatusWidget</class>
   <extends>QWidget</extends>
   <header>statuswidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>RCWidget</class>
   <extends>QWidget</extends>
   <header>rcwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionRemote_Control</sender>
   <signal>toggled(bool)</signal>
   <receiver>rcDock</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>1274</x>
     <y>798</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionRemote_Control</sender>
   <signal>toggled(bool)</signal>
   <receiver>rcDock</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>1125</x>
     <y>777</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
